### DISCUSSION TITLE
ProDiff: performance, training curve, SSIM loss, as well as the distillation part

### DISCUSSION THREAD

**Discussion by unilight** [espnet/espnet#5329] [General]
Hi, first of all, thanks so much for integrating ProDiff into ESPNet! The questions I have are not really issues to be solved so I'm posting them here.
1. How does the current ProDiff implementation compare to other non-AR models like FS2 and JETS?
2. I'm interested in the training curve. Specifically, how many steps does it take to get some intelligible samples?
3. I am also wondering how much SSIM loss helps.
4. Finally, if my understanding is correct, the current implementation only trains the ProDiff teacher right? Have you tried distilling it to a student ProDiff model, or is it not the main goal of this project?
Sorry for the many questions and I hope to hear from you!

**Comment by Fhrozen** [comment]
1. The main difference lies in the decoder. Instead of using a transformer or similar, the decoder employs wavenet for denoise the diffused input. In terms of quality, you can check from the pre-trained models: https://huggingface.co/Fhrozen/tts_prodiff_eng_mspk (or look for prodiff in huggingface)
2. Usually, 800 epochs with 250 iters/epoch and a batchbin of 6M is good. 1200 epochs seem better. but around 400 seems enough for intelligible samples.
3. The original implementation and the ESPnet implementation, both uses SSIM by default, so I did not test without it. But I can say that it seems to speedup the training.
4. Yes, only Prodiff teacher. No, the distilling or student ProDiff is not the main goal. The original implementation shows a improved performance when the model was distilled from 100 steps to 4/2 steps for diffusion. Given that the current model can be trained with only 4 diffusion steps, a distillation seems unneeded, at least for reducing the diffusion steps. Now, you can try reducing the number of parameters. 10 Wavenet Layers seems to work pretty well.
